lib/awful/layout/init.lua

```diff
@@ -417,8 +455,78 @@ function layout.move_handler(c, context, hints) --luacheck: no unused args
     end
 end
 
+-- [UNDOCUMENTED] Handler for `request::restack`.
+--
+-- @signalhandler awful.layout.move_handler
+-- @tparam string context The context
+-- @tparam table hints Additional hints
+-- @tparam[opt=nil] client|nil hints.client The client
+-- @tparam[opt=nil] drawin|nil hints.drawin Additional hints
+function layout._restack_handler(context, hints) -- luacheck: no unused args
+    --TODO Support permissions
+    need_restack = true
+end
+
+--- Arrange the clients on the Z axis.
+-- @staticfct awful.layout.restack
+-- @noreturn
+-- @see arrange
+function layout.restack()
+
+    local layers = {}
+
+    local function append(o)
+        local layer = client_to_layer(o)
+        print(x11_layers_ordered[layer], o)
+        layers[layer] = layers[layer] or {}
+        table.insert(layers[layer], 1, o.drawin and o.drawin or o)
+    end
+
+    local drawins, clients = capi.drawin.get(), capi.client.get(nil, true)
+
+    for _, c in ipairs(clients) do append(c) end
+    for i=#drawins, 1, -1 do append(drawins[i].get_wibox()) end
```

Comment:
Not sure about this, swapping those line breaks quite a few things. For example, the client go on top of the wibar in the "old" builds. If I swap these, the wibox goes on top. That doesn't mean it didn't break anything. It's after all not exactly what it did before. What did you witness being in the wrong order?

Reply:
That must've been a mistake on my end. Wiboxes do indeed go on top when these two lines are swapped. I just thought that it would be the other way around. Didn't notice line 482 inserting into the table at index 1 lol

Reply:
IMO client should go over wibar only if it have `c.ontop == true`